scripts/portstat

```diff
@@ -27,537 +21,25 @@ try:
-                                  format_brate(rates.rx_bps),
-                                  format_prate(rates.rx_pps),
-                                  format_util(rates.rx_bps, port_speed),
-                                  format_number_with_comma(cntr['tx_ok']),
-                                  format_brate(rates.tx_bps),
-                                  format_prate(rates.tx_pps),
-                                  format_util(rates.tx_bps, port_speed)))
-            else:
-                header = header_std
-                if old_cntr is not None:
-                    table.append((key,
-                              self.get_port_state(key),
-                              ns_diff(cntr['rx_ok'], old_cntr['rx_ok']),
-                              format_brate(rates.rx_bps),
-                              format_util(rates.rx_bps, port_speed),
-                              ns_diff(cntr['rx_err'], old_cntr['rx_err']),
-                              ns_diff(cntr['rx_drop'], old_cntr['rx_drop']),
-                              ns_diff(cntr['rx_ovr'], old_cntr['rx_ovr']),
-                              ns_diff(cntr['tx_ok'], old_cntr['tx_ok']),
-                              format_brate(rates.tx_bps),
-                              format_util(rates.tx_bps, port_speed),
-                              ns_diff(cntr['tx_err'], old_cntr['tx_err']),
-                              ns_diff(cntr['tx_drop'], old_cntr['tx_drop']),
-                              ns_diff(cntr['tx_ovr'], old_cntr['tx_ovr'])))
-                else:
-                    table.append((key,
-                              self.get_port_state(key),
-                              format_number_with_comma(cntr['rx_ok']),
-                              format_brate(rates.rx_bps),
-                              format_util(rates.rx_bps, port_speed),
-                              format_number_with_comma(cntr['rx_err']),
-                              format_number_with_comma(cntr['rx_drop']),
-                              format_number_with_comma(cntr['rx_ovr']),
-                              format_number_with_comma(cntr['tx_ok']),
-                              format_brate(rates.tx_bps),
-                              format_util(rates.tx_bps, port_speed),
-                              format_number_with_comma(cntr['tx_err']),
-                              format_number_with_comma(cntr['tx_drop']),
-                              format_number_with_comma(cntr['tx_ovr'])))
-        if table:
-            if use_json:
-                print(table_as_json(table, header))
-            else:
-                print(tabulate(table, header, tablefmt='simple', stralign='right'))
-        if (multi_asic.is_multi_asic() or device_info.is_chassis()) and not use_json:
-            print("\nReminder: Please execute 'show interface counters -d all' to include internal links\n")
+from utilities_common.portstat import Portstat
```

Comment:
what was the reason we have make this as utility `portstat.py`

Reply:
Because our chassis_port_counter_monitor script uses this utility to get counter statistics as well. Overall, I think it is good to decouple the portstat script and the implementation of portstat class since other software may utilize the portstat class in the future.